src/backend/distributed/transaction/backend_data.c

```diff
@@ -781,10 +797,12 @@ AssignDistributedTransactionId(void)
 	uint64 nextTransactionNumber = pg_atomic_fetch_add_u64(transactionNumberSequence, 1);
 	int localGroupId = GetLocalGroupId();
 	TimestampTz currentTimestamp = GetCurrentTimestamp();
+	Oid userId = GetUserId();
```

Comment:
It's kind of ok here because `GetUserId()` only returns a value, but we shouldn't do anything more than necessary while holding a spin lock. 

(we have some instances of calling `GetLocalGroupId()` while holding a spinlock which are a likely source of crashes)